nat-lab/tests/test_network_monitor.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+import time
+from contextlib import AsyncExitStack
+from helpers import setup_mesh_nodes, SetupParameters
+from telio import AdapterType
+from utils.connection_util import ConnectionTag
+
+DEFAULT_WAITING_TIME = 2
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "alpha_setup_params",
+    [
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.DOCKER_SHARED_CLIENT_1,
+                adapter_type=AdapterType.BoringTun,
+            ),
+        ),
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.WINDOWS_VM_1,
+                adapter_type=AdapterType.WireguardGo,
+            ),
+            marks=[pytest.mark.windows],
+        ),
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.MAC_VM,
+                adapter_type=AdapterType.BoringTun,
+            ),
+            marks=[
+                pytest.mark.mac,
+            ],
+        ),
+    ],
+)
+async def test_network_monitor(
+    alpha_setup_params: SetupParameters,
+) -> None:
+    async with AsyncExitStack() as exit_stack:
+        env = await setup_mesh_nodes(exit_stack, [alpha_setup_params])
+        [client_alpha] = env.clients
+
+        time.sleep(DEFAULT_WAITING_TIME)
+        await client_alpha.restart_interface()
+
+        time.sleep(DEFAULT_WAITING_TIME)
```

Comment:
This is blocking sleep on an async function, is intentional?
Despite of that, you can instead do an async `wait_for()` call to `wait_for_log()` with a timeout.

Reply:
Intentional yes for triggers from OS to occur. `wait_for_log()` with a timeout can also be done.